experimental/traffic-portal/src/app/api/user.service.ts

```diff
@@ -267,6 +268,37 @@ export class UserService extends APIService {
 		return this.get<Array<ResponseRole>>(path).toPromise();
 	}
 
+	/**
+	 * Creates a new Role.
+	 *
+	 * @param role The role to create.
+	 * @returns The created role.
+	 */
+	public async createRole(role: RequestRole): Promise<ResponseRole> {
```

Comment:
we can be more specific than this - a response from a POST request to `/roles` has the `PostResponseRole` type. That lets callers use the `lastUpdated` property, which `PUT` responses for whatever reason don't provide.